tests/cli/test_build_and_deploy_vmware.sh

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+# Note: execute this file from the project root directory
+
+#####
+#
+# Make sure we can build an image and deploy it inside vSphere!
+#
+#####
+
+. /usr/share/beakerlib/beakerlib.sh
+
+CLI="./src/bin/composer-cli"
+
+
+rlJournalStart
+    rlPhaseStartSetup
+        if [ -z "$V_HOST" ]; then
+            rlFail "V_HOST is empty!"
+        else
+            rlLogInfo "V_HOST=$V_HOST"
+        fi
+
+        if [ -z "$V_USERNAME" ]; then
+            rlFail "V_USERNAME is empty!"
+        else
+            rlLogInfo "V_USERNAME=$V_USERNAME"
+        fi
+
+        if [ -z "$V_PASSWORD" ]; then
+            rlFail "V_PASSWORD is empty!"
+        else
+            rlLogInfo "V_PASSWORD is configured"
+        fi
+
+        V_DATACENTER="${V_DATACENTER:-RH_Engineering}"
+        rlLogInfo "V_DATACENTER=$V_DATACENTER"
+
+        V_CLUSTER="${V_CLUSTER:-SysMgmt_vMotion}"
+        rlLogInfo "V_CLUSTER=$V_CLUSTER"
+
+        V_NETWORK="${V_NETWORK:-CEE_VM_Network}"
+        rlLogInfo "V_NETWORK=$V_NETWORK"
+
+        V_DATASTORE="${V_DATASTORE:-NFS-Synology-1-RDU}"
+        rlLogInfo "V_DATASTORE=$V_DATASTORE"
+
+        V_FOLDER="${V_FOLDER:-Composer}"
+        rlLogInfo "V_FOLDER=$V_FOLDER"
+
+        if ! rlCheckRpm "python3-pip"; then
+            rlRun -t -c "dnf -y install python3-pip"
+            rlAssertRpm python3-pip
+        fi
+
+        rlRun -t -c "pip3 install pyvmomi"
+
+        SAMPLES="./pyvmomi-community-samples/samples/"
+        if [ ! -d ./pyvmomi-community-samples/ ]; then
+            rlRun -t -c "git clone https://github.com/weldr/pyvmomi-community-samples"
+            pushd ./pyvmomi-community-samples/ && git checkout composer_testing && popd
+        fi
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose start"
+        if [ ! -f ~/.ssh/id_rsa.pub ]; then
+            rlRun -t -c "ssh-keygen -t rsa -N '' -f ~/.ssh/id_rsa"
```

Comment:
If you run this on a local checkout it's going to clobber your ssh key! Better to try and put the artifacts into a /tmp/ directory and pull them from there (the same goes for vmware.toml and the git clone.

Reply:
How come? Isn't this why there's an if-file-doesn't-exist statement above ?

I agree putting the rest of the artifacts in temporary directory but for SSH keys would like to reuse existing keys if possible.

Reply:
> How come? Isn't this why there's an if-file-doesn't-exist statement above ?

Sigh. How did I miss that? Sorry for the noise, looks good to me.